src/utils/query.ts

```diff
@@ -330,3 +332,28 @@ export const parseQueryErrorToString = (error: unknown) => {
 
     return parsedError?.error?.message;
 };
+
+export const DEFAULT_QUERY_SETTINGS = {
+    queryMode: QUERY_MODES.query,
+    transactionMode: TRANSACTION_MODES.implicit,
+    timeout: 60,
+    limitRows: 10000,
+    statisticsMode: STATISTICS_MODES.none,
+    tracingLevel: TRACING_LEVELS.detailed,
+};
+
+export const querySettingsValidationSchema = z.object({
+    timeout: z.preprocess(
+        (val) => (val === '' ? undefined : val),
+        z.coerce.number().positive().default(DEFAULT_QUERY_SETTINGS.timeout),
```

Comment:
timeout never will be undefined, is it ok?
the same question for limitRows.

Reply:
@adameat states these fields are not required

added .optional() check to validation schema

as far as I can see now they are not sent in request and not shown in the dialog box if they are undefined

<img width="870" alt="image" src="https://github.com/user-attachments/assets/ad44fa06-6ebc-47c7-88bb-a6854bd342ac">

Reply:
* to restore schema ofk